### PR TITLE
tests: Fix local testing

### PR DESCRIPTION
**Pull request by l0b0**

### CODE REVIEW

File: test.bash [hunk 1]
```diff
@@ -36,10 +36,10 @@ then
...
 fi
 
-LDI_LINZ_KEY="$1"
-LDI_MFE_KEY="$2"
-LDI_NZDF_KEY="$3"
-LDI_BASEMAPS_KEY="$4"
+export LDI_LINZ_KEY="$1"
+export LDI_MFE_KEY="$2"
+export LDI_NZDF_KEY="$3"
+export LDI_BASEMAPS_KEY="$4"
```
Comment (billgeo) [inline test.bash:42]: I ran test.bash locally but still get those errors.

```
test_elements_exist (test_ldi_ui_elements.UiTest) ... None
Traceback (most recent call last):
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 433, in run
    self.loadUi()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 461, in loadUi
    load_data_err=self.loadAllServices()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 500, in loadAllServices
    service_data_instance.processServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 319, in processServiceData
    self.getServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 296, in getServiceData
    self.getServiceXml()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 361, in getServiceXml
    self.api_key_int.getApiKey(self.domain),
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 74, in getApiKey
    return self.api_keys[domain]
KeyError: 'data.linz.govt.nz'

ok
test_listItem_about_shows_widget_swHelp (test_ldi_ui_elements.UiTest) ... None
Traceback (most recent call last):
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 433, in run
    self.loadUi()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 461, in loadUi
    load_data_err=self.loadAllServices()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 500, in loadAllServices
    service_data_instance.processServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 319, in processServiceData
    self.getServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 296, in getServiceData
    self.getServiceXml()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 361, in getServiceXml
    self.api_key_int.getApiKey(self.domain),
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 74, in getApiKey
    return self.api_keys[domain]
KeyError: 'data.linz.govt.nz'

ok
test_listItem_about_shows_widget_swSettings (test_ldi_ui_elements.UiTest) ... None
Traceback (most recent call last):
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 433, in run
    self.loadUi()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 461, in loadUi
    load_data_err=self.loadAllServices()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 500, in loadAllServices
    service_data_instance.processServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 319, in processServiceData
    self.getServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 296, in getServiceData
    self.getServiceXml()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 361, in getServiceXml
    self.api_key_int.getApiKey(self.domain),
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 74, in getApiKey
    return self.api_keys[domain]
KeyError: 'data.linz.govt.nz'

ok
test_listItem_all_shows_widget_swTableView (test_ldi_ui_elements.UiTest) ... None
Traceback (most recent call last):
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 433, in run
    self.loadUi()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 461, in loadUi
    load_data_err=self.loadAllServices()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 500, in loadAllServices
    service_data_instance.processServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 319, in processServiceData
    self.getServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 296, in getServiceData
    self.getServiceXml()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 361, in getServiceXml
    self.api_key_int.getApiKey(self.domain),
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 74, in getApiKey
    return self.api_keys[domain]
KeyError: 'data.linz.govt.nz'

ok
test_listItem_wfs_shows_widget_swTableView (test_ldi_ui_elements.UiTest) ... None
Traceback (most recent call last):
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 433, in run
    self.loadUi()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 461, in loadUi
    load_data_err=self.loadAllServices()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 500, in loadAllServices
    service_data_instance.processServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 319, in processServiceData
    self.getServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 296, in getServiceData
    self.getServiceXml()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 361, in getServiceXml
    self.api_key_int.getApiKey(self.domain),
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 74, in getApiKey
    return self.api_keys[domain]
KeyError: 'data.linz.govt.nz'

ok
test_listItem_wmts_shows_widget_swTableView (test_ldi_ui_elements.UiTest) ... None
Traceback (most recent call last):
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 433, in run
    self.loadUi()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 461, in loadUi
    load_data_err=self.loadAllServices()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 500, in loadAllServices
    service_data_instance.processServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 319, in processServiceData
    self.getServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 296, in getServiceData
    self.getServiceXml()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 361, in getServiceXml
    self.api_key_int.getApiKey(self.domain),
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 74, in getApiKey
    return self.api_keys[domain]
KeyError: 'data.linz.govt.nz'

ok
test_service_dialog_is_active (test_ldi_ui_elements.UiTest) ... None
Traceback (most recent call last):
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 433, in run
    self.loadUi()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 461, in loadUi
    load_data_err=self.loadAllServices()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 500, in loadAllServices
    service_data_instance.processServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 319, in processServiceData
    self.getServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 296, in getServiceData
    self.getServiceXml()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 361, in getServiceXml
    self.api_key_int.getApiKey(self.domain),
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 74, in getApiKey
    return self.api_keys[domain]
KeyError: 'data.linz.govt.nz'

FAIL
test_sw_tableview_is_default (test_ldi_ui_elements.UiTest) ... None
Traceback (most recent call last):
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 433, in run
    self.loadUi()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 461, in loadUi
    load_data_err=self.loadAllServices()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/linz_data_importer.py", line 500, in loadAllServices
    service_data_instance.processServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 319, in processServiceData
    self.getServiceData()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 296, in getServiceData
    self.getServiceXml()
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 361, in getServiceXml
    self.api_key_int.getApiKey(self.domain),
  File "/root/.local/share/QGIS/QGIS3/profiles/default/python/plugins/linz-data-importer/service_data.py", line 74, in getApiKey
    return self.api_keys[domain]
KeyError: 'data.linz.govt.nz'

ok

======================================================================
ERROR: test_updateServiceDataCache (test_ldi_plugin.UnitLevel)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tests_directory/linz-data-importer/tests/test_ldi_plugin.py", line 367, in test_updateServiceDataCache
    file = files[-1]
IndexError: list index out of range

======================================================================
FAIL: test_handle_corrupt_xml (test_ldi_integration.CorruptXml)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tests_directory/linz-data-importer/tests/test_ldi_integration.py", line 178, in test_handle_corrupt_xml
    self.assertEqual(len(data_types), 2)
AssertionError: 0 != 2

======================================================================
FAIL: test_all_services (test_ldi_integration.UserWorkFlows)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tests_directory/linz-data-importer/tests/test_ldi_integration.py", line 444, in test_all_services
    self.assertEqual(len(data_types), 2)
AssertionError: 0 != 2

======================================================================
FAIL: test_basemaps_filter (test_ldi_integration.UserWorkFlows)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tests_directory/linz-data-importer/tests/test_ldi_integration.py", line 307, in test_basemaps_filter
    self.filter_domain("basemaps.linz.govt.nz")
  File "/tests_directory/linz-data-importer/tests/test_ldi_integration.py", line 342, in filter_domain
    self.assertEqual(self.ldi.dlg.uLabelWarning.text(), '')
AssertionError: 'Error: (basemaps.linz.govt.nz) Invalid API Key' != ''
- Error: (basemaps.linz.govt.nz) Invalid API Key
+ 


======================================================================
FAIL: test_lds_filter (test_ldi_integration.UserWorkFlows)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tests_directory/linz-data-importer/tests/test_ldi_integration.py", line 314, in test_lds_filter
    self.filter_domain("data.linz.govt.nz")
  File "/tests_directory/linz-data-importer/tests/test_ldi_integration.py", line 342, in filter_domain
    self.assertEqual(self.ldi.dlg.uLabelWarning.text(), '')
AssertionError: 'Error: (basemaps.linz.govt.nz) Invalid API Key' != ''
- Error: (basemaps.linz.govt.nz) Invalid API Key
+ 


======================================================================
FAIL: test_should_filter_wfs_by_map_bbox (test_ldi_integration.UserWorkFlows)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tests_directory/linz-data-importer/tests/test_ldi_integration.py", line 321, in test_should_filter_wfs_by_map_bbox
    self.filter_data_by_map("NZ Primary Parcels")
  File "/tests_directory/linz-data-importer/tests/test_ldi_integration.py", line 459, in filter_data_by_map
    self.assertEqual(self.ldi.dlg.uLabelWarning.text(), '')
AssertionError: 'Error: (basemaps.linz.govt.nz) Invalid API Key' != ''
- Error: (basemaps.linz.govt.nz) Invalid API Key
+ 


======================================================================
FAIL: test_wfs_import (test_ldi_integration.UserWorkFlows)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tests_directory/linz-data-importer/tests/test_ldi_integration.py", line 293, in test_wfs_import
    self.import_service("wfs")
  File "/tests_directory/linz-data-importer/tests/test_ldi_integration.py", line 392, in import_service
    self.assertEqual(self.ldi.dlg.uLabelWarning.text(), '')
AssertionError: 'Error: (basemaps.linz.govt.nz) Invalid API Key' != ''
- Error: (basemaps.linz.govt.nz) Invalid API Key
+ 


======================================================================
FAIL: test_wmts_import (test_ldi_integration.UserWorkFlows)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tests_directory/linz-data-importer/tests/test_ldi_integration.py", line 300, in test_wmts_import
    self.import_service("wmts")
  File "/tests_directory/linz-data-importer/tests/test_ldi_integration.py", line 392, in import_service
    self.assertEqual(self.ldi.dlg.uLabelWarning.text(), '')
AssertionError: 'Error: (basemaps.linz.govt.nz) Invalid API Key' != ''
- Error: (basemaps.linz.govt.nz) Invalid API Key
+ 


======================================================================
FAIL: test_service_dialog_is_active (test_ldi_ui_elements.UiTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tests_directory/linz-data-importer/tests/test_ldi_ui_elements.py", line 68, in test_service_dialog_is_active
    self.assertTrue(self.ldi.dlg.isVisible())
AssertionError: False is not true

----------------------------------------------------------------------
Ran 40 tests in 46.440s

FAILED (failures=8, errors=1)
Finished running test linz-data-importer.tests.run_tests.run_test_modules.
```

Comment (billgeo) [reply test.bash:42]: It looks to me that the domain and API keys aren't being added to the plugin settings for some reason.

Comment (l0b0) [reply test.bash:42]: That's weird, it works for me locally. Which commit are you testing on?

Comment (billgeo) [reply test.bash:42]: Ignore that comment. I had an issue with teh arguments.

**Review by billgeo — APPROVED** [review]

**Pull request merged**